File: clients/ruby/lib/etwin/hammerfest/hammerfest_server.rb
# typed: strict
# frozen_string_literal: true

module Etwin
  module Hammerfest
    # A valid Hammerfest server
    class HammerfestServer < T::Enum
      extend T::Sig

      enums do
        HammerfestEs = new('hammerfest.es')
        HammerfestFr = new('hammerfest.fr')
        HfestNet = new('hfest.net')
      end

      sig { returns(String) }
      def to_s
        T.cast(serialize, String)
      end

      sig { returns(String) }
      def inspect
        "HammerfestServer(#{self})"
      end
    end
  end
end

File: packages/core/src/lib/hammerfest/hammerfest-server.ts
import { $Ucs2String } from "kryo/lib/ucs2-string";
import { WhiteListType } from "kryo/lib/white-list";

/**
 * A Hammerfest server.
 */
export type HammerfestServer = "hammerfest.fr" | "hfest.net" | "hammerfest.es";

export const $HammerfestServer: WhiteListType<HammerfestServer> = new WhiteListType({
  itemType: $Ucs2String,
  values: ["hammerfest.fr", "hfest.net", "hammerfest.es"],
});
